lib/src/main/java/ru/nsu/fit/XMLSerialization/XMLSerializator.java

```diff
@@ -0,0 +1,129 @@
+package ru.nsu.fit.XMLSerialization;
+
+import java.lang.reflect.Field;
+import java.lang.reflect.Type;
+import java.util.*;
+import java.util.stream.Stream;
+import javax.xml.parsers.DocumentBuilder;
+import javax.xml.parsers.DocumentBuilderFactory;
+import javax.xml.transform.*;
+import javax.xml.transform.dom.DOMSource;
+import javax.xml.transform.stream.StreamResult;
+
+import org.w3c.dom.Document;
+import org.w3c.dom.Element;
+
+public class XMLSerializator {
+  private int id = 0;
+  private static final List<String> primAndWrappers =
+      Arrays.asList(
+          "byte",
+          "short",
+          "int",
+          "long",
+          "float",
+          "double",
+          "boolean",
+          "char",
+          "java.lang.Byte",
+          "java.lang.Short",
+          "java.lang.Integer",
+          "java.lang.Long",
+          "java.lang.Float",
+          "java.lang.Double",
+          "java.lang.Boolean",
+          "java.lang.Character",
+          "java.lang.String");
+
+  public void createXMLDocument(Stream<Object> inputStream, StreamResult stream) {
+    DocumentBuilderFactory factory = DocumentBuilderFactory.newInstance();
+    DocumentBuilder builder;
+    try {
+      builder = factory.newDocumentBuilder();
+
+      Document doc = builder.newDocument();
+      Element rootElement = doc.createElementNS("xml_steam", "XML_STEAM");
+
+      Element objectPull = doc.createElement("Object_pull");
+      Element objectSteam = doc.createElement("Object_steam");
+      doc.appendChild(rootElement);
+      rootElement.appendChild(objectSteam);
+      rootElement.appendChild(objectPull);
+
+      TransformerFactory transformerFactory = TransformerFactory.newInstance();
+      Transformer transformer = transformerFactory.newTransformer();
+      Map<Object, Map.Entry<Integer, String>> parsedObjects = new IdentityHashMap<>();
```

Comment:
<Object, Integer>?

Reply:
yes, it's absolutely useless string, fixed